### DISCUSSION TITLE
IPullable &   WithOffsetResetPolicy(AutoOffsetReset.Earliest)

### DISCUSSION THREAD

**Discussion by MikeKenyonAccuris** [tomasfabian/ksqlDB.RestApi.Client-DotNet#79] [Q&A]
We are scanning a KSQL table using a pull query and   `.WithOffsetResetPolicy(AutoOffsetReset.Earliest)`.

In upgrading to version 6.0 from 5.1, looks like `CreateQueryStream<>()`got replaced with `CreatePullQuery<>()`, but an IPullable doesn't support the `.WithOffsetResetPolicy()` modifier.  

How is one supposed to do this post the 6.0 change?

**Comment by tomasfabian** [comment]
Hello @MikeKenyonAccuris,
in version 6.0  `CreateQueryStream<>()` got replaced with `CreatePushQuery<>()` not with `CreatePullQuery<>()`.
